plugin/ochttp/server_test.go

```diff
@@ -273,3 +276,88 @@ func TestHandlerProxiesHijack_HTTP2(t *testing.T) {
 		t.Errorf("got = %q; want = %q", g, w)
 	}
 }
+
+func TestEnsureTrackingResponseWriterSetsStatusCode(t *testing.T) {
+	// Ensure that the trackingResponseWriter always sets the spanStatus on ending the span.
+	// Because we can only examine the Status after exporting, this test roundtrips a
+	// couple of requests and then later examines the exported spans.
+	// See Issue #700.
+	spanInspector := new(spanExporter)
+	trace.RegisterExporter(spanInspector)
+	defer trace.UnregisterExporter(spanInspector)
+
+	tests := []struct {
+		res  *http.Response
+		want trace.Status
+	}{
+		{res: &http.Response{StatusCode: 200}, want: trace.Status{0, `"OK"`}},
+		{res: &http.Response{StatusCode: 500}, want: trace.Status{2, `"UNKNOWN"`}},
+		{res: &http.Response{StatusCode: 403}, want: trace.Status{7, `"PERMISSION_DENIED"`}},
+		{res: &http.Response{StatusCode: 401}, want: trace.Status{16, `"UNAUTHENTICATED"`}},
+		{res: &http.Response{StatusCode: 429}, want: trace.Status{8, `"RESOURCE_EXHAUSTED"`}},
+	}
+
+	for i, tt := range tests {
```

Comment:
Name these tests instead and create subtests.

Reply:
I don't follow, could you please help me with more wording/guidance on what you mean?

Reply:
I was suggesting to create subtests such as https://github.com/rakyll/opencensus-go/blob/master/plugin/ochttp/propagation/tracecontext/propagation_test.go#L74. It makes it possible to run each test individually and generates a pretty output.